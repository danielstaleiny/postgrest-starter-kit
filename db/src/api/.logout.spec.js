const ROOT = process.cwd()
const resetdb = require(`${ROOT}/util/reset-db.js`)
const { get, post, put, patch, del } = require(`${ROOT}/util/fetch.js`)(
  'http://localhost:3000/'
)

describe('/rpc/logout', () => {
  test('success 200 without refresh_token', async () => {
    const res = await post('rpc/logout')({
      withRole: { id: 1, role: 'webuser' },
    })
    expect(res).toMatchSnapshot()
  })

  test('success 200 with refresh_token', async () => {
    const res = await post('rpc/logout')({
      withRole: { id: 1, role: 'webuser' },
      body: {
        refresh_token: 'de688895-6181-440f-a25a-8a9ba14b2162',
      },
    })
    expect(res).toMatchSnapshot()
  })

  test('success 200 with refresh_token which is not in db', async () => {
    const res = await post('rpc/logout')({
      withRole: { id: 1, role: 'webuser' },
      body: {
        refresh_token: 'de688895-6181-440f-a25a-000000000000',
      },
    })
    expect(res).toMatchSnapshot()
  })
})
