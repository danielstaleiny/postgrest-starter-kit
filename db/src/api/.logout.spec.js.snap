// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/rpc/logout success 200 with refresh_token 1`] = `
Object {
  "body": "{\\"ok\\" : true}",
  "status": 200,
  "statusText": "OK",
}
`;

exports[`/rpc/logout success 200 without refresh_token 1`] = `
Object {
  "body": "{\\"ok\\" : true}",
  "status": 200,
  "statusText": "OK",
}
`;
